crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -199,35 +199,38 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// Checks for exception handlers that catch non-exception classes.
+    /// Checks for `raise` statements and exception handlers
+    /// that raise or handle non-exception classes.
     ///
     /// ## Why is this bad?
-    /// Catching classes that do not inherit from `BaseException` will raise a TypeError at runtime.
+    /// Attempts to raise or catch classes that do not inherit from `BaseException`
+    /// will cause a `TypeError` to be raised at runtime.
     ///
     /// ## Example
     /// ```python
     /// try:
-    ///     1 / 0
+    ///     raise int from str
     /// except 1:
     ///     ...
     /// ```
     ///
     /// Use instead:
     /// ```python
     /// try:
-    ///     1 / 0
-    /// except ZeroDivisionError:
+    ///     raise ZeroDivisionError from RuntimeError
+    /// except ArithmeticError:
     ///     ...
     /// ```
     ///
     /// ## References
-    /// - [Python documentation: except clause](https://docs.python.org/3/reference/compound_stmts.html#except-clause)
+    /// - [Python documentation: The `raise` statement](https://docs.python.org/3/reference/simple_stmts.html#raise)
+    /// - [Python documentation: `except` clause](https://docs.python.org/3/reference/compound_stmts.html#except-clause)
     /// - [Python documentation: Built-in Exceptions](https://docs.python.org/3/library/exceptions.html#built-in-exceptions)
     ///
     /// ## Ruff rule
     ///  This rule corresponds to Ruff's [`except-with-non-exception-classes` (`B030`)](https://docs.astral.sh/ruff/rules/except-with-non-exception-classes)
-    pub(crate) static INVALID_EXCEPTION_CAUGHT = {
-        summary: "detects exception handlers that catch classes that do not inherit from `BaseException`",
+    pub(crate) static INVALID_EXCEPTION = {
```

Comment:
Seeing the changes to the documentation, I now think that these should be two error codes: `invalid-exception-caught` and `invalid-exception-raised`

Reply:
Yeah, I agree that this would be clearer.

Reply:
Two or three separate codes? On the raising side, there's the raised exception, and there's the cause: the `Foo` and the `e` in `raise Foo from e`. I think if we split it out, we kind of have to go to three, or else one of the two will still have an awkward name: `invalid-exception-raised` doesn't make sense for a bad "cause", because that's not the exception raised.

Reply:
I think it would be fine to have two codes: `invalid-exception-caught` and `invalid-raise-statement`. One code would be about exception handlers, the other would be about `raise` statements.

Reply:
Sounds reasonable. Maybe just `invalid-raise`? Not sure `-statement` adds value.